From a788f0062897efca0762772d26f209ac5d91c9c6 Mon Sep 17 00:00:00 2001
From: Peter Ujfalusi <peter.ujfalusi@ti.com>
Date: Thu, 13 Jun 2013 15:01:39 +0200
Subject: [PATCH 07/39] ASoC: core: soc-fw: Fix soc_fw_dapm_graph_load()

Issue: GMINL-3679

We need to clear the route struct before we use it to avoid random data in
important items.
This will fix a kernel dump when trying to read the DAPM via debugfs.

Change-Id: Ieaa7ca5b57db9c503ca2d4df07c8b2e3d69ca215
Signed-off-by: Peter Ujfalusi <peter.ujfalusi@ti.com>
---
 sound/soc/soc-fw.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/sound/soc/soc-fw.c b/sound/soc/soc-fw.c
index ee0fbc3..dbf54ca 100644
--- a/sound/soc/soc-fw.c
+++ b/sound/soc/soc-fw.c
@@ -698,6 +698,8 @@ static int soc_fw_dapm_graph_load(struct soc_fw *sfw,
 
 	dev_dbg(sfw->dev, "ASoC: adding %d DAPM routes\n", count);
 
+	memset(&route, 0, sizeof(route));
+
 	for (i = 0; i < count; i++) {
 		elem = (struct snd_soc_fw_dapm_graph_elem *)sfw->pos;
 		sfw->pos += sizeof(struct snd_soc_fw_dapm_graph_elem);
-- 
1.9.1

